### PR TITLE
Fethullah week 18 project mongo api

### PR DESCRIPTION
**Pull request by FD000**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Endpoints: 
+1) .../books - Get access to all data in our collection
+2) .../books/:memberId - Search book by id
+3) .../books/name/:authorName - Search and find book by author name
+4) .../books/minpage/:pages - Search and find book by filtering minimum page the books have
+5) .../books/maxpage/:pages - Search and find book by filtering maximum page the books have
+6) .../books/isbn/:isbn - Search and find book by isbn number
+7) .../highestrated - List top 20 highest rated books
```
Comment (KarolCislowski) [inline README.md:12]: nice documentation 👍🏻

File: server copy 2.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  authors: "Fethullah Dincer",
+  average_rating: 10,
+  isbn: 102030,
+  isbn13: 405060,
+  language_code: "SE",
+  num_pages: 500,
+  ratings_count: 5050566060,
+  text_reviews_count: 87820206 
+})
+newAuthor.save() */
```
Comment (KarolCislowski) [inline server copy 2.js:48]: remember to clean your code from unused code.

File: server copy 2.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+import booksData from './data/books.json'
+// import netflixData from './data/netflix-titles.json'
+// import topMusicData from './data/top-music.json'
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.Promise = Promise
+
+const bookSchema = new mongoose.Schema({ // #1
+  bookID: Number,
```
Comment (KarolCislowski) [inline server copy 2.js:20]: you do not need to pass the old id because mongoose will provide a new one.

File: server copy 2.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+// Add middlewares to enable cors and json body parsing
+app.use(cors())
+app.use(bodyParser.json())
+
+// Start defining your routes here
+app.get('/', (req, res) => {
+  res.send('Hello world')
+})
+
+let myAPI
```
Comment (KarolCislowski) [inline server copy 2.js:82]: why you created your variable outside the endpoint?

File: server copy 2.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+// #5 Max Page filter
+app.get('/books/maxpage/:pages', async (req,res) => {
+  const { pages } = req.params;
+
+  myAPI = await Book.find({ num_pages: { $lte: pages}});
+
+  res.json({length: myAPI.length, data: myAPI})
+})
+
+// #6 Ratings filter (GÖR en select)
```
Comment (KarolCislowski) [inline server copy 2.js:152]: remember to only use English in your code.

File: server copy 2.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+  res.json({length: myAPI.length, data: myAPI})
+})
+
+
+
+// #4 Min page filter
+app.get('/books/minpage/:pages', async (req,res) => {
+  const { pages } = req.params;
+
+  myAPI = await Book.find({ num_pages: { $gte: pages}});
```
Comment (KarolCislowski) [inline server copy 2.js:138]: 👍🏻

File: server copy 2.js [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+  res.json({length: myAPI.length, data: myAPI})
+})
+
+// #5 Max Page filter
+app.get('/books/maxpage/:pages', async (req,res) => {
+  const { pages } = req.params;
+
+  myAPI = await Book.find({ num_pages: { $lte: pages}});
```
Comment (KarolCislowski) [inline server copy 2.js:147]: 👍🏻

File: server copy 2.js [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+  try {
+    myAPI = await Book.findById(memberId);
+    if(myAPI) {
+      res.json({length: myAPI.length, data: myAPI})
+    } else {
+      res.status(404).json({ error: 'Not found'});
+    }
+  } catch {
+    res.status(400).json({ error: 'Invalid request'})
```
Comment (KarolCislowski) [inline server copy 2.js:122]: nice error handeling.

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

File: server.js [hunk 9]
```diff
@@ -32,8 +48,91 @@ app.get('/', (req, res) => {
...
+  const { authorName } = req.params;
+  myAPI = await Book.find(
+     {authors: {$regex: authorName}});
+  res.json({length: myAPI.length, data: myAPI})
+})
+
+// #4 Minimum page filter
+app.get('/books/minpage/:pages', async (req,res) => {
+  const { pages } = req.params;
+  myAPI = await Book.find({ num_pages: { $gte: pages}});
```
Comment (tuttibalutti) [inline server.js:86]: Cool that you use the aggregation pipeline operator👌

**Review by tuttibalutti — CHANGES_REQUESTED** [review]
Great job, Fethullah! I really liked your API and the documentation of it🤩  It seems like you have three different versions of you code: server copy.js, server copy 2.js and the server.js (which I guess is the right one?). I suggest that you remove those copies from your folder to minimize future confusion👀  Well done!

**Pull request closed without merging**